### DISCUSSION TITLE
Test if reference exists or not empty

### DISCUSSION THREAD

**Discussion by dsaintilma-flinks** [cue-lang/cue#1527] [Q&A] [answered]
Hello.

Simple question: I would like to populate a list of only declared variables so I would not get the following errors:
```
$ cue genfiles ./configs/dev/app01
objectSets: reference "service" not found:
    ./kube_tool.cue:6:2
objectSets: reference "statefulSet" not found:
    ./kube_tool.cue:8:2
objectSets: reference "daemonSet" not found:
    ./kube_tool.cue:9:2
```

Here's the content of the objectSets:
```
package kube

objects: [ for v in objectSets for x in v { x } ]

objectSets: [
	service,
	deployment,
	statefulSet,
	daemonSet,
	// configMap,
	secret,
    application,
]
```
Is there a way to achieve this?

Thanks in advance for the help.

**Comment by verdverm** [comment, marked as answer]
I think your best bet will be to filter for when they have a value, but you will still need to have the name in scope. You can do this like so:

```cue
objects: [ for v in objectSets if v != _|_ for x in v { x } ]

service: _
deployment: _
... one for each

objectSets: [ ... as above ... ]
```

- The `if v != _|_` in the list comprehension, this is the filter. This will change to `must()` once #575 / #943 are implemented.
- Then we introduce all of the supported list elements as `<thing>: _` so that we are not referring to unknown labels.

**Reply by dsaintilma-flinks** [reply]
Yep, this works. I tried to utilize `if x != _|_` but for each elements in the `objectSets` inside structs.

Thanks!
